src/zarr/storage/object_store.py

```diff
@@ -158,12 +168,13 @@ async def _transform_list_dir(
     # We assume that the underlying object-store implementation correctly handles the
     # prefix, so we don't double-check that the returned results actually start with the
     # given prefix.
-    prefix_len = len(prefix)
+    prefix_len = len(prefix) + 1  # If one is not added to the length, all items will contain "/"
```

Comment:
Actually, should we test if a user defines prefix as (e.g.,) "foo/" rather than "foo" before automatically adding 1?

Reply:
Ideally it shouldn't be necessary to check _again_ in Python